core/src/test/java/org/elasticsearch/action/support/replication/TransportReplicationActionTests.java

```diff
@@ -678,6 +680,91 @@ protected ReplicaResult shardOperationOnReplica(Request request) {
         assertIndexShardCounter(0);
     }
 
+    /**
+     * This test ensures that replication operations adhere to the {@link IndexMetaData#WAIT_FOR_ACTIVE_SHARDS_SETTING} setting.
+     */
+    public void testChangeWaitForActiveShardsSetting() throws Exception {
+        final String indexName = "test";
+
+        // test negative wait_for_active_shards value fails
+        expectThrows(IllegalArgumentException.class, () ->
```

Comment:
True, I already have negative tests for this setting, so its a bit redundant to test it with respect to TRA here. I've changed the tests accordingly.